### PR TITLE
Origin/user friendly searching

### PR DESCRIPTION
**Pull request by EdblackGitHub**
**Description:** Added Manual user-friendly option for our code to search CCVs

**Issue:** Adding a User-friendly option to the CCVs searching

**Dependencies:** List dependencies on open pull requests, other issue/work tickets, etc.

**Merge Deadline:** Merge today 11/15/2024

**Pre-Merge Checklist:**

- [ ] I have ensured this code follows the style guidelines of this project.
- [X] I have performed a comprehensive self-review of the code.
- [X] I have included comments clarifying hard-to-understand areas of the code.
- [ ] I have updated the documentation corresponding to my change(s).
- [ ] I have confirmed the code executes properly on my local environment.
- [ ] I have confirmed all new/modified data models are structured correctly.
- [ ] I have included all necessary database migration files.
- [ ] I have verified that all new and existing tests pass.
- [ ] I have received approval from the requisite number of reviewers.
- [ ] I have squashed all working commits into as few final commits as reasonably possible.
- [ ] I have confirmed Github reports all automated tests/checks are passing.

**Post-Merge Checklist:**

- [ ] Delete working branch @ Github repository
- [ ] Pull latest master branch to local repository
- [ ] Delete working branch @ local repository

### CODE REVIEW

File: app/uid/views.py [hunk 1]
```diff
@@ -9,6 +9,81 @@
...
 
+import os
+from .forms import SearchForm
+import requests
+import urllib.parse
+
+# Neo4j connection details
+NEO4J_USERNAME = os.getenv('NEO4J_USERNAME', 'neo4j')        # Default username if env var not set
+NEO4J_PASSWORD = os.getenv('NEO4J_PASSWORD', 'password')      # Default password if env var not set
+NEO4J_HOST = os.getenv("NEO4J_HOST", "localhost")  # Default to localhost if not set
```
Comment (vbhayden) [inline app/uid/views.py:20]: This whole block shoul be removed -- get the connection string from settings.

```py
from django.conf import settings

connection_url = settings.DATABASE_URL
```

**Comment by vbhayden** [comment]
@EdblackGitHub has this been merged with main recently?  The UID generation stuff in `uid/views.py` looks dated.

**Comment by EdblackGitHub** [comment]
I did do a git pull to get the latest changes, after I made the branch from main.

**Comment by vbhayden** [comment]
So the most recent version of that views file is here:

https://github.com/adlnet/LDSS-XSS/blob/main/app/uid/views.py#L60

and the corresponding line in most recent version of this branch is showing up as:

https://github.com/adlnet/LDSS-XSS/blob/origin/User_Friendly_searching/app/uid/views.py#L232

Your local one might be behind, if you tried to merge just the local branch without retrieving the most recent remote main.

File: app/uid/models.py [hunk 2]
```diff
@@ -21,6 +22,79 @@ def check_neo4j_connection():
...
+     #   """Override the save method to automatically link the alias to a NeoTerm if context is provided."""
+      #  super(Alias, self).save(*args, **kwargs)
+
+       # if self.context:  # If context is provided, find the NeoTerm by name and link it
+        #    term = NeoTerm.nodes.get_or_none(name=self.context)
+         #   if term:
+          #      self.link_to_term(term)
+
+
+class Alias(StructuredNode):
```
Comment (vbhayden) [inline app/uid/models.py:64]: This class looks like it's more related to terms yeah?  Why is it in the UID portion of the project etc.

File: docker-compose.yml [hunk 3]
```diff
@@ -47,25 +48,17 @@ services:
...
-      SENTENCE_TRANSFORMERS_HOME: "${SENTENCE_TRANSFORMERS_HOME}"
-      HF_HOME: "${HF_HOME}"
-      TRANSFORMERS_CACHE: "${TRANSFORMERS_CACHE}"
-      HF_DATASETS_CACHE: "${HF_DATASETS_CACHE}"
+      #NEO4J_URL: "bolt://neo4j:password@neo4j:7687"
     volumes:
       - ./app:/opt/app/openlxp-xss
-      - huggingface_data:/opt/app/huggingface
     depends_on:
-      - neo4j
```
Comment (vbhayden) [inline docker-compose.yml:58]: The neo4j dependency should not be removed here, as the app container does definitely depend on it.

File: app/uid/models.py (hunk 2 above)
Comment (EdblackGitHub) [reply app/uid/models.py:64]: They are in UID folder as I asked Florian if he had an ideal location or a preference for the search function location and he said I could leave it in the UID folder.

File: docker-compose.yml (hunk 3 above)
Comment (EdblackGitHub) [reply docker-compose.yml:58]: Uhh, I did not touch the .yml file for the manual search function.

**Comment by vbhayden** [comment]
Just merged in main into this one locally and things did update accordingly, so maybe try again on your end?

For the compose file, the main's contents for that block are currently:
```
    depends_on:
      db-xss:
        condition: service_healthy
      neo4j:
        condition: service_healthy
```
which are a good bit different, yeah?  Double check that you did the merge order correctly, or this will end up being a big soup sandwich when it hits main.

**Pull request closed without merging**